app/display/representation-javafx/src/main/java/org/csstudio/display/builder/representation/javafx/widgets/AutoCompleteRepresentation.java

```diff
@@ -0,0 +1,597 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void unregisterListeners() {
+        model_widget.propWidth().removePropertyListener(styleChangedListener);
+        model_widget.propHeight().removePropertyListener(styleChangedListener);
+        model_widget.propForegroundColor().removePropertyListener(styleChangedListener);
+        model_widget.propBackgroundColor().removePropertyListener(styleChangedListener);
+        model_widget.propFont().removePropertyListener(styleChangedListener);
+        model_widget.propPlaceholder().removePropertyListener(styleChangedListener);
+        model_widget.runtimePropValue().removePropertyListener(contentChangedListener);
+        model_widget.propItemsFromPV().removePropertyListener(contentChangedListener);
+        model_widget.propItems().removePropertyListener(contentChangedListener);
+        model_widget.propEnabled().removePropertyListener(enableChangedListener);
+        model_widget.runtimePropPVWritable().removePropertyListener(enableChangedListener);
+        super.unregisterListeners();
+    }
+
+    /**
+     * Confirm selected value and write it to PV or local variable.
+     *
+     * @param value Selected value by user.
+     */
+    private void confirmValue(final String value) {
+        if (!model_widget.runtimePropPVWritable().getValue()) {
+            return;
+        }
+
+        isUserTyping = false;
+
+        if (model_widget.propConfirmDialog().getValue()) {
+            final String message = model_widget.propConfirmMessage().getValue();
+            final String password = model_widget.propPassword().getValue();
+            if (password.length() > 0) {
+                if (toolkit.showPasswordDialog(model_widget, message, password) == null) {
+                    return;
+                }
+            } else if (!toolkit.showConfirmationDialog(model_widget, message)) {
+                return;
+            }
+        }
+
+        try {
+            final VType currentPvValue = model_widget.runtimePropValue().getValue();
+            Object mappedValue;
+
+            if (currentPvValue instanceof VEnum) {
+                mappedValue = FormatOptionHandler.parse(currentPvValue, value,
+                    FormatOption.DEFAULT);
+            } else {
+                try {
+                    if (value.contains(".")) {
+                        mappedValue = Double.parseDouble(value);
+                    } else {
+                        mappedValue = Integer.parseInt(value);
+                    }
+                } catch (NumberFormatException e) {
+                    VType convertedValue = VType.toVType(value);
+                    mappedValue = (convertedValue != null) ? convertedValue : value;
+                }
+            }
+
+            toolkit.fireWrite(model_widget, mappedValue);
+        } catch (Exception e) {
+            System.err.println("Error writing to PV: " + e.getMessage());
+            e.printStackTrace();
```

Comment:
Instead of println and e.printStackTrace, use logger:
```
import static org.csstudio.display.builder.representation.ToolkitRepresentation.logger;
...
logger.log(Level.WARNING, "Error writing to PV", e);
```